lms/djangoapps/courseware/courses.py

```diff
@@ -94,9 +93,39 @@ def get_course_with_access(user, action, course_key, depth=0, check_if_enrolled=
     check_if_enrolled: If true, additionally verifies that the user is either enrolled in the course
       or has staff access.
     """
-    assert isinstance(course_key, CourseKey)
```

Comment:
Was there a reason for this assertion?  Did it provide a different or better error, or was it pointless?